lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,71 @@
+
+## Overview
+
+We will utilize two separate internal networks to isolate our production services from slave nodes that will be executing tests.  The two networks will be connected by a L3 router which will provide connectivity between internal networks as well outbound connectivity to the internet.  Security groups will be used to limit connectivity from the outside world, allow connectivity between hosts on the same network, and prohibit connectivity from slave nodes to the rest of the infrastructure.  This network topology will be configured in Neutron via Ansible automation and will be scoped to production project.  It should be trivial for developers to recreate this topology within their own tenant and project.
+
+### Projects
+
+Network resources will be created within a `control-plane` project.  We will be utilizing shared networks, which must be created by a user with the `cloud_admin` role on that project. Compute resources for control plane services will be created within the `control-plane` project.  Nodepool itself will be configured to use a specific `nodepool` project for creating its instances with interfaces attached to a shared network that was created on the `control-plane` project.
+
+### Networks
+
+We will create two separate internal networks and hosts will be given interfaces on them depending on the role of said host.
+
+* `control-plane 192.168.10.0/24` - This network will be used for communication between all of our production services, as well as automation between the bastion host and our producion servers.  As per our current infrastructure, the following static hosts will have interfaces on this network and internal DNS updated accordingly:
+  * bastion.bonncyi*internal.portbleu.com
+  * logs.bonnyci*internal.portbleu.com
+  * merger01.bonnyci*internal.portbleu.com
+  * nodepool.bonncyi*internal.portbleu.com
+  * zuul.bonnyci*internal.portbleu.com
+* `nodepool 10.0.0.0/8` - This is an shared network that will be used by nodepool slaves. None of the static service hosts within the control plane will receive an interface on this network.
+
+Our cloud is also configured with an external network named `external` that provides us currently with 30 floating IPs.  These will be assigned to hosts depending on their need, see [Floating IPs](#floating-ips).
+
+### Routers
+
+To allow connectivity between the two isolated networks, a single L3 router will be created.  The router will have a router interface attached for both the `control-plane` and `nodepool` networks.  Additionally, it will have its router gateway set to provide outbound connectivity through the external network and allow for floating IP association.
```

Comment:
Can you add some detail on how the nodepool slaves will use the router to access the merger(s)? Will this router be configured as the default gateway for the nodepool slaves?